src/HealthGPS/analysis_module.cpp

```diff
@@ -26,44 +26,38 @@ SimulationModuleType AnalysisModule::type() const noexcept {
-                                  [](const auto &a, const auto &b) {
-                                      return a.get_risk_factor_value(factor2) <
-                                             b.get_risk_factor_value(factor2);
-                                  });
-
-    auto min_factor2 = min_max.first->get_risk_factor_value(factor2);
-    auto max_factor2 = min_max.second->get_risk_factor_value(factor2);
-
-    // The dimensions of the 2D map is the number of integer values of each factor, or 100 bins of
-    // equal size, whichever is smaller (100 is an arbitrary number, it could be any other number
-    // depending on the desired resolution of the map)
-    auto factor1_bins = std::min(100, static_cast<int>(max_factor1 - min_factor1));
-    auto factor2_bins = std::min(100, static_cast<int>(max_factor2 - min_factor2));
-
-    // Create a 2D map with the specified number of bins
-    calculated_factors_ =
-        std::unordered_map<int, std::unordered_map<int, std::unordered_map<std::string, double>>>{};
-    for (int i = 0; i < factor1_bins; i++) {
-        for (int j = 0; j < factor2_bins; j++) {
-            calculated_factors_[i][j] = std::unordered_map<std::string, double>{};
-        }
+    std::vector<core::Identifier> factors{"Gender"_id, "Age"_id};
+
+    std::vector<int> factor_bins;
+
+    for (const auto &factor : factors) {
+        auto min_max = std::minmax_element(context.population().cbegin(), context.population().cend(),
+                                           [&factor](const auto &a, const auto &b) {
+                                               return a.get_risk_factor_value(factor) <
+                                                      b.get_risk_factor_value(factor);
+                                           });
+
+        auto min_factor = min_max.first->get_risk_factor_value(factor);
+        auto max_factor = min_max.second->get_risk_factor_value(factor);
+
+        // The number of bins to use for each factor is the number of integer values of the factor, or 100 bins of
+        // equal size, whichever is smaller (100 is an arbitrary number, it could be any other number
+        // depending on the desired resolution of the map)
+        factor_bins.push_back(std::min(100, static_cast<int>(max_factor - min_factor)));
     }
+
+    // The number of factors to calculate is the number of factors minus the length of the `factors` vector.
+    auto num_factors_to_calc = context.mapping().entries().size() - factors.size();
+
+    // The product of the number of bins for each factor can be used to calculate the size of the `calculated_factors_` in the next step
+    auto total_num_bins = std::accumulate(factor_bins.cbegin(), factor_bins.cend(), 1,
+                                    std::multiplies<int>());
+
+    // Set the vector size and initialise all values to 0.0
+    calculated_factors_.resize(total_num_bins * num_factors_to_calc, 0.0);
```

Comment:
Zero is the default value anyway, so it's fine to just do this:

```suggestion
    calculated_factors_.resize(total_num_bins * num_factors_to_calc);
```